sdk/core/azure-core/inc/azure/core/io/body_stream.hpp

```diff
@@ -148,12 +149,8 @@ namespace Azure { namespace Core { namespace IO {
      */
     explicit MemoryBodyStream(const uint8_t* data, int64_t length) : m_data(data), m_length(length)
     {
-      if (!data || length < 0)
-      {
-        throw std::invalid_argument(
-            "Length cannot be negative: " + std::to_string(length)
-            + ", and the data pointer cannot be null.");
-      }
+      AZURE_ASSERT_MSG(
+          data && length >= 0, "Length cannot be negative, and the data pointer cannot be null.");
```

Comment:
Why isn't this another case of `data != nullptr || length == 0`? It's weird that we allow the empty range `[ptr, ptr + 0)` when `ptr` is non-null but reject it when `ptr` is null.